plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useTinyEditor } from '../hooks/useTinyEditor';
+import MagicIcon from '../../assets/images/icons/magic.svg';
+
+type WooApiMessage = {
+	role: string;
+	content: string;
+};
+
+type WooApiResponse = {
+	generated_text: string;
+	previous_messages: WooApiMessage[];
+};
+
+const generatingContentPhrases = [
+	__( 'Please wait…', 'woocommerce' ),
+	__( 'Just a little longer…', 'woocommerce' ),
+	__( 'Almost there…', 'woocommerce' ),
+];
+
+const getGeneratingContentPhrase = () =>
+	generatingContentPhrases[
+		Math.floor( Math.random() * generatingContentPhrases.length )
+	];
+
+const DESCRIPTION_MAX_LENGTH = 300;
+
+export function WriteItForMeButtonContainer() {
+	const [ fetching, setFetching ] = useState( false );
+	const generatingContentPhraseInterval = useRef< number >();
+	const titleEl = useRef< HTMLInputElement >(
+		document.querySelector( '#title' )
+	);
+	const [ productTitle, setProductTitle ] = useState< string >(
+		titleEl.current?.value || ''
+	);
+	const tinyEditor = useTinyEditor();
+
+	useEffect( () => {
+		titleEl.current?.addEventListener( 'keyup', ( e ) =>
+			setProductTitle( ( e.target as HTMLInputElement ).value || '' )
+		);
+	}, [ titleEl ] );
+
+	useEffect( () => {
+		if ( fetching ) {
+			tinyEditor.setContent( getGeneratingContentPhrase() );
+			generatingContentPhraseInterval.current = setInterval(
+				() => tinyEditor.setContent( getGeneratingContentPhrase() ),
+				2000
+			);
+		} else {
+			clearInterval( generatingContentPhraseInterval.current );
+		}
+	}, [ fetching ] );
+
+	const buildPrompt = () => {
+		const instructions = [
+			`Write a product description with the following product title: "${ productTitle }."`,
+			'Use a 9th grade reading level.',
+			`Make the description ${ DESCRIPTION_MAX_LENGTH } words or less.`,
+			'Structure the description using standard HTML paragraph, strong and list tags.',
+			'Do not include a heading at the very top of the description.',
+		];
```

Comment:
Might be worth adding a fail-safe to the prompt

"If you are unable to write a useful description based on the product title you simply respond with an apology that you are unable to do so based on the information provided."

Reply:
Interesting thought, I'll be curious to see how that works in practice. I'll add it for now 👍🏻

Reply:
On initial testing the model is far to eager to apologize in it's results. I may leave this for a later follow-up when we iterate on the prompt.

Reply:
Sounds good, I was on the fence about it. Thanks for testing it out!